tools/isledecomp/isledecomp/compare/core.py

```diff
@@ -286,31 +287,26 @@ def _add_match_in_array(
         data_type = self.cv.types.keys[node.data_type.key.lower()]
 
         if data_type["type"] == "LF_ARRAY":
-            array_size_bytes = data_type["size"]
 
-            array_type = self.cv.types.get(data_type["array_type"])
+            array_element_type = self.cv.types.get(data_type["array_type"])
 
-            assert array_type.size is not None
-            array_type_size = array_type.size
+            assert node.data_type.members is not None
 
-            array_length, modulus = divmod(array_size_bytes, array_type_size)
-            assert modulus == 0
-
-            for i in range(array_length):
-                orig_element_base_addr = orig_addr + i * array_type_size
-                recomp_element_base_addr = recomp_addr + i * array_type_size
-                if array_type.members is None:
+            for i, array_element in enumerate(node.data_type.members):
```

Comment:
If this is an array, `member.name` should be `[offset]`. If so, you don't need the enumerate. You should update the f-strings too in case the name would ever get used by the sanitize code.

Reply:
Not sure I understand exactly what you mean. In the "array of scalars" case, the parameter `name` is something like `g_unk0x100f3218` and `array_element.name` is e.g. `[0]`. Do you still want to join the strings to `g_unk0x100f3218[0]`?

Similarly, if we have an array of structs, we find e.g. `name=g_actorLODs`, `array_element.name=[0]`, and `member.name=m_name`. Would `g_actorLODs[0].m_name` still be your desired outcome here? I am not sure I understand your point regarding the sanitize code.

In any case, I'll start by refactoring the code based on these suggestions.

Reply:
Yeah, both of those examples look correct to me. The connection with the sanitize code has to do with the fact that we are adding things to the program database. We have had functions that refer to an address that is not the start of a variable, and sanitize will want to find the name for whatever lives at that address. That name would now be `g_unk0x100f3218[1]` instead of `g_unk0x100f3218+4`, for example.